File: golang-devildyw/simple/L104二叉树的最大深度.go
package main

func main() {

}

func maxDepth(root *TreeNode) int {
	// 将大的问题拆解为子问题 通过递归解决 需要找到最小子问题 并根据其进行自下而上的演进
	if root == nil {
		return 0
	}

	return max(maxDepth(root.Left), maxDepth(root.Right)) + 1
}

File: java-devildyw/src/main/java/top/dyw/leetcode/Hot100/L104二叉树的最大深度.java
package top.dyw.leetcode.Hot100;

public class L104二叉树的最大深度 {
    public int maxDepth(TreeNode root) {
        if (root==null) {
            return 0;
        }

        return Math.max(maxDepth(root.right), maxDepth(root.left)) + 1;
    }
}
